osu.Game/Screens/Select/Carousel/CarouselBeatmap.cs

```diff
@@ -77,5 +82,7 @@ public override int CompareTo(FilterCriteria criteria, CarouselItem other)
         }
 
         public override string ToString() => Beatmap.ToString();
+
+        public void UpdateRank(ScoreInfo score) => rank.Value = score?.Rank;
```

Comment:
Very weird - you're exposing a readonly `Rank` bindable that can't be changed and then a method to *mutate* the bindable. Just make the bindable publicly settable?